service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -70,22 +79,105 @@ private void run3037(CaseDetails caseDetails) {
         caseDetails.setData(data);
     }
 
-    private void run3093(CaseDetails caseDetails) {
+    private void run3080(CaseDetails caseDetails) {
         CaseData caseData = getCaseData(caseDetails);
 
         List<Element<AdditionalApplicationsBundle>> additionalApplicationsBundle = caseData
             .getAdditionalApplicationsBundle();
 
-        if (caseData.getAdditionalApplicationsBundle().size() < 1) {
+        Element<AdditionalApplicationsBundle> firstBundle = additionalApplicationsBundle.get(0);
+        if (firstBundle.getId().equals(FIRST_BUNDLE_ID)) {
+
+            firstBundle.getValue().setC2DocumentBundle(firstBundle.getValue()
+                .getC2DocumentBundle()
+                .toBuilder()
+                .supportingEvidenceBundle(buildSupportingEvidenceBundle())
+                .document(DocumentReference.builder()
+                    .filename("S45C-921052410100.pdf")
+                    .url("http://dm-store-prod.service.core-compute-prod.internal/documents/5ddafb9c-1396-44c1-a4dc-25204b0989f0")
+                    .binaryUrl("http://dm-store-prod.service.core-compute-prod.internal/documents/5ddafb9c-1396-44c1-a4dc-25204b0989f0/binary")
+                    .build())
+                .build());
+
+            additionalApplicationsBundle.set(0, firstBundle);
+
+        } else {
             throw new IllegalStateException(String
-                .format("Migration failed on case %s: Case has %s additional applications",
-                    caseData.getFamilyManCaseNumber(), additionalApplicationsBundle.size()));
+                .format("Migration failed on case %s: Expected %s but got %s",
+                    caseData.getFamilyManCaseNumber(), FIRST_BUNDLE_ID, firstBundle.getId()));
         }
 
-        additionalApplicationsBundle.remove(0);
+        Element<AdditionalApplicationsBundle> secondBundle = additionalApplicationsBundle.get(1);
 
-        Map<String, Object> data = caseDetails.getData();
+        if (secondBundle.getId().equals(SECOND_BUNDLE_ID)) {
+            swapC2WithSupportingDocument(secondBundle.getValue(), additionalApplicationsBundle);
+        } else {
+            throw new IllegalStateException(String
+                .format("Migration failed on case %s: Expected %s but got %s",
+                    caseData.getFamilyManCaseNumber(), SECOND_BUNDLE_ID, secondBundle.getId()));
+        }
 
+        Map<String, Object> data = caseDetails.getData();
         data.put("additionalApplicationsBundle", additionalApplicationsBundle);
     }
+
+    private List<Element<SupportingEvidenceBundle>> buildSupportingEvidenceBundle() {
+        SupportingEvidenceBundle firstBundle = SupportingEvidenceBundle.builder()
```

Comment:
the risk is that user could remove supporting doc via manage doc event, so restoring will recreate wrong state, but, we can accept risk